common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java

```diff
@@ -221,26 +235,39 @@ public B setClaims(String claims) {
             return self();
         }
 
-        public B setRequestHeaders(HashMap<String, String> requestHeaders){
+        public B setRequestHeaders(HashMap<String, String> requestHeaders) {
             mRequestHeaders = requestHeaders;
             return self();
         }
 
         public Builder<B> setWebViewZoomEnabled(boolean webViewZoomEnabled) {
-            this.webViewZoomEnabled = webViewZoomEnabled;
+            mWebViewZoomEnabled = webViewZoomEnabled;
             return self();
         }
 
         public Builder<B> setWebViewZoomControlsEnabled(boolean webViewZoomControlsEnabled) {
-            this.webViewZoomControlsEnabled = webViewZoomControlsEnabled;
+            mWebViewZoomControlsEnabled = webViewZoomControlsEnabled;
+            return self();
+        }
+
+        public Builder<B> setCpInstallationDetail(final @NonNull Context context) {
+            try {
+                // Do not use COMPANY_PORTAL_APP_PACKAGE_NAME, since we actually want this to be
+                // CP's package name - not brokerHost's - even in localDebug mode.
+                final PackageInfo packageInfo =
+                        context.getPackageManager().getPackageInfo("com.microsoft.windowsintune.companyportal", 0);
+                mCpVersion = packageInfo.versionName;
+            } catch (final PackageManager.NameNotFoundException e) {
+                // CP is not installed. No need to do anything.
```

Comment:
Should we take charge of logging this here, so we can put in the logs the package name we tried to launch?

Reply:
I intentionally avoid logging here because otherwise this will spam the logs.